### PR TITLE
Review & merge slf/doc-name-game2

### PR DESCRIPTION
**Pull request by slfritchie**
This is a hodge-podge PR.
1. Docs clarification, especially `name-game-sketch.org`
2. Small PB definition changes.
3. Add CoC namespace and locator args to high & low PB APIs.

Unit tests are passing for me (with exception of rare repair EQC test failure (`doublewrite_diff`,which is known problem)), and dialyzer output appears to be clean.

### CODE REVIEW

**Comment by kuenishi** [comment]
There seems to be no changes in low level PB APIs while it is mentioned in this pull request summary. My question is in what client routing is being done? I thought it was to be at flu proxy client. But Flu server also have to be aware of locator and namespaces, to make sure file names are unique?

**Comment by slfritchie** [comment]
Short answer: there is no client routing yet ... it still needs writing.

There is a change in the LL part of the PB protocol: https://github.com/basho/machi/pull/50/files#diff-310df0484f27882f8d9a22ac5b041a1eL379.  It's quite possible that I have overlooked something in the LL client library, though?

**Comment by kuenishi** [comment]
Umm sorry! My eyes were just holes and not reading anything.

File: src/machi_util.erl [hunk 1]
```diff
@@ -134,34 +146,45 @@ make_projection_filename(DataDir, File) ->
...
--spec parse_filename( Filename :: string() ) -> [ string() ].
+-spec parse_filename( Filename :: string() ) -> {} | {string(), machi_dt:coc_namespace(), machi_dt:coc_locator(), string(), string() }.
 parse_filename(Filename) ->
     case string:tokens(Filename, "^") of
-        [_Prefix, _UUID, _SeqNo] = L -> L;
-        _ -> []
+        [Prefix, CoC_NS, CoC_Loc, UUID, SeqNo] ->
+            {Prefix, CoC_NS, list_to_integer(CoC_Loc), UUID, SeqNo};
+        [Prefix,          CoC_Loc, UUID, SeqNo] ->
+            %% string:tokens() doesn't consider "foo^^bar" as 3 tokens {sigh}
```
Comment (jadeallenx) [inline src/machi_util.erl:170]: Sometimes perl5 `split` seems pretty nice :)

**Comment by jadeallenx** [comment]
I haven't tested this yet. I just wanted to read through the code and get a feel for the changes.  I didn't see anything that struck me as crazy though.

**Comment by jadeallenx** [comment]
:+1: 

tests pass and dialyzer looks clean.

**Pull request merged**